pyplatypus/segmentation/generator.py

```diff
@@ -91,25 +91,46 @@ def __init__(
         self.only_images = only_images
         self.net_h = net_h
         self.net_w = net_w
+        self.ensemble_net_h = ensemble_net_h
+        self.ensemble_net_w = ensemble_net_w
         self.h_splits = h_splits
         self.w_splits = w_splits
-        self.channels = channels if isinstance(channels, list) else [channels]
+        self.channels = channels
+        self.is_ensemble = self.check_model_is_ensemble()
         self.augmentation_pipeline = augmentation_pipeline
         self.batch_size = batch_size
         self.shuffle = shuffle
         self.subdirs = subdirs
         self.column_sep = column_sep
-        self.target_size = (net_h, net_w)
         self.classes = len(colormap)
         self.return_paths = return_paths
-        self.config = self.create_images_masks_paths(self.path, self.mode, self.only_images, self.subdirs,
-                                                     self.column_sep)
+        self.config = create_images_masks_paths(self.path, self.mode, self.only_images, self.subdirs, self.column_sep)
         self.indexes = None
         self.steps_per_epoch = self.calculate_steps_per_epoch()
         print(len(self.config["images_paths"]), "images detected!")
         print("Set 'steps_per_epoch' to:", self.steps_per_epoch)
         self.on_epoch_end()
 
+    def check_model_is_ensemble(self) -> bool:
```

Comment:
JUst leaving a note for myself to move it all to pydantic.

Reply:
ok